chia/wallet/wallet.py

```diff
@@ -363,15 +362,26 @@ async def _generate_unsigned_transaction(
             )
         elif exclude_coins is not None:
             raise ValueError("Can't exclude coins when also specifically including coins")
+
         assert len(coins) > 0
         self.log.info(f"coins is not None {coins}")
         spend_value = sum([coin.amount for coin in coins])
-
+        self.log.info(f"spend_value is {spend_value} and total_amount is {total_amount}")
         change = spend_value - total_amount
         if negative_change_allowed:
             change = max(0, change)
-
-        assert change >= 0
+        if change < 0 < fee:
```

Comment:
I think I'm missing something. perhaps I don't understand this syntax. But `change` has to be < 0 for this condition to be true, right? Is that used as a place-holder to ask the wallet to pick coins to spend? And why do we not allow this path for zero-fee transactions?

Did we have a problem here before, where we would sometimes fail to spend because of `MINTING_COIN`?

Reply:
this should only kick in if we're missing the fee and coins were not selected in this method but passed through. This happens when we generate offers as trade_manager only selects coins that are needed to complete the offer, not fee. But you have a point that this condition could be clearer.